pkg/cluster/compute/affinityrule/affinityrule.go

```diff
@@ -29,20 +39,25 @@ func NewAffinityRule(instance *wmi.WmiInstance) (*AffinityRule, error) {
 }
 
 // CreateAffinityRule
-func CreateAffinityRule(whost *host.WmiHost, name string, ruleType int) (affinityRule *AffinityRule, err error) {
-	query := "SELECT * FROM meta_class WHERE __CLASS = 'MSCluster_AffinityRule'"
-	classes, err := instance.GetWmiClasssesFromHostRawQuery(whost, string(constant.FailoverCluster), query)
+// Make sure to call Close once done using this instance
```

Comment:
Is this the pattern followed commonly, to return a disposable instance in Create?

Reply:
pattern is derived from availability set. Currently returned value is not used by caller i.e. it is immediately disposed.